src/System.Management.Automation/FormatAndOutput/common/StringDecorated.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+
+namespace System.Management.Automation.Internal
+{
+    /// <summary>
+    /// Extensions to String type to calculate and render decorated content.
+    /// </summary>
+    public class StringDecorated
```

Comment:
This class is used extensively in output/formatting. We will create an object of it for every string output. We don't have measurement, but I guess it would be a lot GC pressure, especially most of the objects will be transient.

I suggest we have a struct for the same functionality that will be used internally, and a public class of it for external uses.

Reply:
An example of this in .NET is `struct ValueStringBuilder` they use internally to avoid `StringBuilder`.